src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -0,0 +1,130 @@
+package org.broadinstitute.dsde.firecloud.core
+
+import akka.actor.{Actor, Props}
+import akka.event.Logging
+import akka.pattern.pipe
+
+import org.broadinstitute.dsde.firecloud.core.ProfileClient.UpdateProfile
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.service.{UserService, FireCloudRequestBuilding}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+
+import spray.client.pipelining._
+import spray.http.HttpResponse
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.routing.RequestContext
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object ProfileClient {
+  case class UpdateProfile(userInfo: UserInfo, profile: Profile)
+  def props(requestContext: RequestContext): Props = Props(new ProfileClientActor(requestContext))
+}
+
+class ProfileClientActor(requestContext: RequestContext) extends Actor with FireCloudRequestBuilding {
+
+  implicit val system = context.system
+  import system.dispatcher
+  val log = Logging(system, getClass)
+
+  override def receive: Receive = {
+
+    case UpdateProfile(userInfo: UserInfo, profile: Profile) =>
+      val parent = context.parent
+      val pipeline = authHeaders(requestContext) ~> sendReceive
```

Comment:
oooh if that's not thread-safe then I have a problem in AgoraPermissionActor if not elsewhere :(